sdk/src/Services/CognitoSync/Custom/SyncManager/Dataset.cs

```diff
@@ -691,12 +691,15 @@ private void SynchornizeInternal(Runtime.AsyncOptions options)
 
             //make sure we have the latest identity id
 #if BCL35 || UNITY
-            CognitoCredentials.GetIdentityId();
+            try
```

Comment:
This will cause build to fail on BCL45 and PCL platforms. You need to have the try block above the `#if BCL35 || UNITY` code. also you can get rid of the line 700 & 701.